### PR TITLE
optimize: reduce the db and network request when undoLog  or lockKey is empty

### PR DESCRIPTION
**Pull request by lightClouds917**
### Ⅱ. Does this pull request fix one issue?

[issue](https://github.com/seata/seata/issues/2388)
optimize: unnecessary to register branch transaction when undo is empty

### CODE REVIEW

**Comment by lightClouds917** [comment]
@zjinlei  PTAL,THS

File: rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java [hunk 1]
```diff
@@ -211,23 +211,32 @@ private void processLocalCommitWithGlobalLocks() throws SQLException {
...
         } catch (Throwable ex) {
-            LOGGER.error("process connectionProxy commit error: {}", ex.getMessage(), ex);
-            report(false);
-            throw new SQLException(ex);
+            if (hasUndoLog) {
+                LOGGER.error("process connectionProxy commit error: {}", ex.getMessage(), ex);
+                report(false);
+                throw new SQLException(ex);
+            } else {
+                LOGGER.error("process connectionProxy commit error: {},and the undolog is null", ex.getMessage(), ex);
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java:236]: lack throw new SQLException(ex);

Comment (lightClouds917) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java:236]: ok,i will fix

File: rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java [hunk 2]
```diff
@@ -211,24 +211,30 @@ private void processLocalCommitWithGlobalLocks() throws SQLException {
...
-            register();
-        } catch (TransactionException e) {
-            recognizeLockKeyConflictException(e, context.buildLockKeys());
-        }
-
-        try {
-            if (context.hasUndoLog()) {
+        if (context.hasUndoLog()) {
+            try {
+                register();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java:216]: I think it's better to put it in registers and flushUndoLogs.
if there is no undolog why flushUndoLogs method execution without judgment

Comment (lightClouds917) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java]: I have optimized the code, PTAL

Comment (lightClouds917) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java]: I have finish it.PTAL

**Pull request merged**